### ISSUE TITLE
What is correct way to rotate sddm.log file?

### ISSUE THREAD

**Issue by dsh2dsh** [sddm/sddm#1874]
Hi,

what is correct way to rotate `sddm.log` file?

**Comment by tusooa** [comment]
TLDR: `truncate --size 0 sddm.log`

Recently I have experienced a problem where my sddm user session log taking up a huge part of space on my computer. Merely deleting the log files will not free up space, as it turns out that the file descriptor is still being used, and thus the file is not removed from the filesystem although there are no longer any paths that points to it.

I then looked into logrotate, which on my Gentoo installation comes with this default configuration for sddm:

```
] cat /etc/logrotate.d/sddm
/var/log/sddm.log {
	missingok
	notifempty
	copytruncate
}
```

`copytruncate` means to first copy the log file, and then truncate it to size 0. This way, the log file being used is not deleted, but the file length is set to 0, and any new writes to the currently-opened fd will write from the beginning of the file again.

**Comment by dsh2dsh** [comment]
> TLDR: `truncate --size 0 sddm.log`

Thank you, @tusooa On FreeBSD I use

```
/var/log/sddm.log			644  7	   *	720   EJNp
```

in `/etc/newsyslog.conf`. The key option is `p`, which indicates that the zero-th rotated file should not be compressed. I think this issue is resolved now and I'm closing it.
